mods/afkkick/init.lua

```diff
@@ -15,65 +15,72 @@ local players = {}
 local checkTimer = 0
 
 minetest.register_on_joinplayer(function(player)
-	local playerName = player:get_player_name()
-	if mc_helpers.checkPrivs(player) then
-		-- player is teacher, do not time or kick
-		players[playerName] = nil
-	else
-		players[playerName] = {
-			lastAction = minetest.get_gametime()
-		}
-	end
+    local playerName = player:get_player_name()
+    if mc_helpers.checkPrivs(player) then
+        -- player is teacher, do not time or kick
+        players[playerName] = nil
+    else
+        players[playerName] = {
+            lastAction = minetest.get_gametime()
+        }
+    end
 end)
 
 minetest.register_on_leaveplayer(function(player)
-	local playerName = player:get_player_name()
-	players[playerName] = nil
+    local playerName = player:get_player_name()
+    players[playerName] = nil
 end)
 
 minetest.register_on_chat_message(function(playerName, message)
-	players[playerName]["lastAction"] = minetest.get_gametime()
+    players[playerName]["lastAction"] = minetest.get_gametime()
 end)
 
 minetest.register_globalstep(function(dtime)
-	local currGameTime = minetest.get_gametime()
 
-	--Check for inactivity once every CHECK_INTERVAL seconds
-	checkTimer = checkTimer + dtime
 
-	local checkNow = checkTimer >= CHECK_INTERVAL
-	if checkNow then
-		checkTimer = checkTimer - CHECK_INTERVAL
-	end
+    -- Optional dependency on mc_worldManager so we don't kick players while spawn is generating
+    if (mc_worldManager ~= nil and mc_worldManager.SpawnGenerated() ~= nil) then
```

Comment:
Don't you need `teacher` priv to generate spawn? Teachers are not kicked by `afkkick`. Is there another place we need to ensure that the proper player on_join is given `teacher`?

Reply:
Spawn is initially generated before any privileges can/are assigned to a player. This isn't an issue with small spawns (like the current vancouver_osm), but during testing I had the world auto-generate spawn as UBC. Because schematic placement  locked up chat commands (single thread shenanigans), players get autokicked before the player can grant themselves the command. This change was mainly for singleplayer to prevent the spawn from not fully spawning on first launch.